cfripper/config/rule_configs/allow_http_ports_open_to_world.py

```diff
@@ -0,0 +1,27 @@
+from cfripper.config.filter import Filter
+from cfripper.model.enums import RuleMode
+
+"""
+To use this Filter, or any Filter, make sure to include it in the `Config` instantiation.
+
+```python
+FILTERS = [allow_http_ports_open_to_world_rules_config_filter]
+
+config = Config(
+    ...
+    rules_filters=FILTERS,
+)
+```
+"""
+
+allow_http_ports_open_to_world_rules_config_filter = Filter(
+    reason="It can be acceptable to have Security Groups publicly available on ports 80 or 443.",
+    rule_mode=RuleMode.ALLOWED,
+    eval={
+        "and": [
+            {"exists": {"ref": "open_ports"}},
+            {"or": [{"in": [80, {"ref": "open_ports"}]}, {"in": [443, {"ref": "open_ports"}]}]},
+        ]
+    },
+    rules={"EC2SecurityGroupOpenToWorldRule", "EC2SecurityGroupIngressOpenToWorldRule"},
+)
```

Comment:
Right now, we are adding one big error with all open ports. This filter would allow also port 9090 to be open and the issue would be hidden.
We should revert and add one error per open port or preprocess filters before?
I'm not sure about the best option here.
What do you think?

Reply:
Right, good point.

I think we still need to remove the hardcoded values - alerting on these ports may be valuable in some cases. This filter will not work as intended, so should be removed from the PR.

One error per port is not scalable - preprocessing of filters I also think is not the best way. Perhaps filters should be created as exemptions, rather than ignoring by default? Wdyt?

Reply:
What about adding a new kind of post-processors?
Suppose filters are post-processors because they are modifying the output. We could create a new kind of post-processors to group errors.

Right now, we do:
- Analyse
- Inside the security group, we pre-process the error before adding it. So we create one big error.
- We filter errors

Proposal:
- Analyse
- We filter errors
- We post-process the errors

In each error, we have the context, so we have to take the worst severity or even let the aggregator decide. 
The current use case would be as easy as moving the group logic from the rule to the new aggregator

Reply:
We could do, I'm hesitant we are designing for an edge case here. I think it's something we can consider in the future perhaps, even add as an issue on the repo if you think it's worth it. 😄 

Thinking about the filter again, instead of using the `in` logic, which leads to false negatives, we could use `eq` against the powerset (essentially all combinations of subsets). That way, if you use ports 79 and 80, the filter will fail, but if you use ports 80 and 443, it will be ok. This approach should be scalable unless you need to allow more than 20 ports, in which case the computation involved might be more hassle than the simplicity of the filter. I've added a unit test to show this working.

Only additional issue in this approach is the error message will not filter out the 'allowed' ports. We would need to refactor the rule + use the post processing aggregator you mentioned.